src/app/shared/helpers/null-undefined-predicates.ts

```diff
@@ -17,3 +17,7 @@ export function ensureDefined<T>(value: T | undefined, message = 'expected value
   if (isNotUndefined(value)) return value;
   throw new Error(message);
 }
+
+export function assertNotUndefined<T>(value: T | undefined, message = 'expected value not to be undefined'): asserts value is T {
+  if (!isDefined(value)) throw new Error(message);
+}
```

Comment:
Q1: It's almost the same function as `ensureDefined` but its purpose is different, I thought it was 2 separate functions but this is debatable. What do you think ?

Q2: The functions `assertDefined` and `ensureDefined` are not aligned with the file name because they are not predicates, they throw errors. Is there a more suited file for that or should I create one like `assert.ts` ?

Reply:
Q1. yes, looks good as 2 functions. 
As I said when discussing about it, I am not sure "assert" is a good prefix as in some prog lang assertion can be disabled. `must...` may be an alternative as we did it in Go.
Q2. Yes, probably they would be better in a different file.

Reply:
Fixed
I'll move `ensureDefined` in another PR